frontend/src/concepts/pipelines/content/StartingStatusModal.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import {
+  Modal,
+  ModalBody,
+  ModalHeader,
+  ModalFooter,
+  Spinner,
+  Stack,
+  StackItem,
+  Tab,
+  Tabs,
+  TabTitleText,
+  Panel,
+  PanelMain,
+  List,
+  ListItem,
+  Button,
+} from '@patternfly/react-core';
+import { usePipelinesAPI } from '#~/concepts/pipelines/context';
+import { getStatusFromCondition } from '#~/concepts/pipelines/content/utils.tsx';
+import K8sStatusIcon from '#~/concepts/pipelines/content/K8sStatusIcon.tsx';
+import { K8sCondition } from '#~/k8sTypes.ts';
+import './StartingStatusModal.scss';
+
+const PROGRESS_TAB = 'Progress';
+const EVENT_LOG_TAB = 'Events log';
+
+const notReadySection = (
+  <div className="pipeline-status-modal__not-ready-section">
+    <p />
+    <div>
+      This may take a while. You can close this modal and continue using the application. The
+      pipeline server will be available when initialization is complete.
+    </div>
+  </div>
+);
+
+type StartingStatusModalProps = {
+  onClose: () => void;
+  conditionLog: K8sCondition[];
+};
+
+// because you can close and re-open this modal; and the 'status' tab only needs to show the current conditions;
+// but the 'event log' tab should show everything (from before the user opens the modal)
+// need to get the events from the parent that already/always exists.
+// (this component only exists when it is open): eslint rules do not allow this component to exist if it not open
+const StartingStatusModal: React.FC<StartingStatusModalProps> = ({ onClose, conditionLog }) => {
+  const { pipelinesServer } = usePipelinesAPI();
+  const [activeTab, setActiveTab] = React.useState<string>(PROGRESS_TAB);
+  const isServerReadyAndCompletelyDone = pipelinesServer.crStatus?.conditions?.some(
+    (c) => c.type === 'Ready' && c.status === 'True',
+  );
+
+  const spinner = (
+    <div className="pipeline-status-modal__spinner-container">
+      <div className="pipeline-status-modal__spinner-text">Initializing Pipeline Server</div>
+      <Spinner size="lg" />
+    </div>
+  );
+
+  const renderProgress = () => (
+    <Stack hasGutter>
+      <StackItem>
+        <Stack hasGutter>
+          {pipelinesServer.crStatus?.conditions?.map((condition, index) => {
+            const containerStatus = getStatusFromCondition(condition);
+            return (
+              <StackItem key={`${condition.type}-${index}`}>
+                <div className="pipeline-status-modal__status-item">
```

Comment:
what are these classnames. are you using them for tests? if so then use data-testid

Reply:
those are gone; using flex-items now.

Reply:
removed; using flex items